shop/src/main/java/ro/msg/learning/shop/controller/impl/ProductControllerImpl.java

```diff
@@ -0,0 +1,70 @@
+package ro.msg.learning.shop.controller.impl;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.controller.ProductController;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.dto.ProductToSaveDto;
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.utils.mapper.ProductMapper;
+
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/products")
+public class ProductControllerImpl implements ProductController {
+    private static final String ID = "id";
+    private static final String CREATE_SUCCESS = "Product successfully created";
+    private static final String DELETE_SUCCESS = "Product successfully deleted";
+    private static final String UPDATE_SUCCESS = "Product successfully updated";
+    private final ProductService productService;
+
+    public ProductControllerImpl(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping
+    @Override
+    public List<ProductDto> findAllProducts() {
+        return productService.findAllProducts().stream()
+                .map(ProductMapper::ProductToDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/{id}")
+    public ProductDto findById(@PathVariable(ID) Integer id) {
+        return ProductMapper.ProductToDto(productService.findById(id));
+    }
+
+    @PostMapping
+    public ResponseEntity<String> createProduct(@RequestBody ProductToSaveDto productDto) {
+        productService.createProduct(ProductMapper.DtoToSaveToProduct(productDto,
```

Comment:
Again, the mappings should be done within the service. The controller should onhly call create product and pass the `productDto` and any other necessary data